homeassistant/components/overkiz/binary_sensor.py

```diff
@@ -35,70 +35,81 @@ class OverkizBinarySensorDescription(
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
+        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
     ),
     # WaterSensor/WaterDetectionSensor
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_WATER_DETECTION,
         name="Water",
         icon="mdi:water",
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
+        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
     ),
     # AirSensor/AirFlowSensor
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_GAS_DETECTION,
         name="Gas",
         device_class=BinarySensorDeviceClass.GAS,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
+        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
     ),
     # OccupancySensor/OccupancySensor
     # OccupancySensor/MotionSensor
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_OCCUPANCY,
         name="Occupancy",
         device_class=BinarySensorDeviceClass.OCCUPANCY,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.PERSON_INSIDE),
+        value_fn=lambda state: state == OverkizCommandParam.PERSON_INSIDE,
     ),
     # ContactSensor/WindowWithTiltSensor
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_VIBRATION,
         name="Vibration",
         device_class=BinarySensorDeviceClass.VIBRATION,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
+        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
     ),
     # ContactSensor/ContactSensor
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_CONTACT,
         name="Contact",
         device_class=BinarySensorDeviceClass.DOOR,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.OPEN),
+        value_fn=lambda state: state == OverkizCommandParam.OPEN,
     ),
     # Siren/SirenStatus
     OverkizBinarySensorDescription(
         key=OverkizState.CORE_ASSEMBLY,
         name="Assembly",
         device_class=BinarySensorDeviceClass.PROBLEM,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.OPEN),
+        value_fn=lambda state: state == OverkizCommandParam.OPEN,
     ),
     # Unknown
     OverkizBinarySensorDescription(
         key=OverkizState.IO_VIBRATION_DETECTED,
         name="Vibration",
         device_class=BinarySensorDeviceClass.VIBRATION,
-        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
+        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
+    ),
+    # DomesticHotWaterProduction/WaterHeatingSystem
+    OverkizBinarySensorDescription(
+        key=OverkizState.IO_OPERATING_MODE_CAPABILITIES,
+        name="Energy Demand Status",
+        device_class=BinarySensorDeviceClass.HEAT,
+        value_fn=_get_value_for_io_operation_mode_capabilities,
     ),
```

Comment:
Looks like a new sensor was added (maybe cherry picked in from other other branch by accident?).  Please put this in a new PR.

Reply:
You are right, I will remove it. I added it to try if the type casting was working, since this is the only binary_sensor that uses a dict value (.get), where all the others use a string comparison. 

This sensor got omitted from the original PR, because the typing was a mess.

Reply:
Removed in 72afc94.